test/ex_doc/retriever_test.exs

```diff
@@ -121,6 +121,16 @@ defmodule ExDoc.RetrieverTest do
     end
   end
 
+  test "docs_from_modules warns when module was not compiled with flag --docs" do
+    module = ExDoc.RetrieverTest.ModuleNotCompiledWithFlagDocs
+    {:ok, ^module} = :compile.file('#{__DIR__}/retriever_test/#{module}.erl')
+    "Elixir." <> module_name_suffix = module |> to_string
+
+    assert ExUnit.CaptureIO.capture_io(:stderr, fn ->
```

Comment:
Why not add this as an import at the beginning of the file and then just call `capture_io` here?

```elixir
import ExUnit.CaptureIO
```